### PR TITLE
Add detailed info to offer trace

### PR DESCRIPTION
**Pull request by njgheorghita**
### What was wrong?
Updated `trace_offer` to return more detail about what happened during the `Offer`, specifically to identify between cases where the offered key was rejected, or the transfer failed, which is helpful to know inside the state bridge.

### How was it fixed?
- added `OfferTrace` type to capture more information about failed `trace_offer` requests

### To-Do

[//]: # (Stay ahead of things, add list items here!)
- [ ] Clean up commit history and use [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/).

### CODE REVIEW

**Comment by njgheorghita** [comment]
Blocked by #1485, since that's updating the `content_keys` type used here, but this is ready for review, and I'll make the adjustments once that gets merged

**Review by KolbyML — APPROVED** [review]
:shipit:

**Comment by carver** [comment]
> Blocked by #1485, since that's updating the `content_keys` type used here, but this is ready for review, and I'll make the adjustments once that gets merged

merged

File: ethportal-api/src/types/portal_wire.rs
Comment (carver) [inline ethportal-api/src/types/portal_wire.rs:560]: "Rejected" sounds to me a bit like the peer thinks we offered something in an invalid way. To me "Declined" sounds more like they just don't want the data we offered.

Also, a quick docstring would be nice. A draft for you:
```suggestion
    /// Peer is not interested in any of the offered data
    Declined,
    /// This offer failed, perhaps locally or from a timeout or transfer failure
    Failed,
```

**Review by carver — APPROVED** [review]
LGTM, 🚢 after the master rebase

File: ethportal-api/src/types/portal_wire.rs
Comment (njgheorghita) [reply ethportal-api/src/types/portal_wire.rs]: Nice, gonna start applying this mindset to the women who don't want to go out on a date with me...

**Pull request merged**
